plugins/download.py

```diff
@@ -131,6 +132,10 @@ def _get_packages(self, pkg_specs, source=False):
                 queries.append(func(pkg_spec))
             except dnf.exceptions.PackageNotFoundError as e:
                 logger.error(dnf.i18n.ucd(e))
+                if self.base.conf.strict:
+                    logger.error(_("Exiting due to strict setting.")
```

Comment:
better. It should `raise dnf.exceptions.Error(e)`. (Plugins should not end with `sys.exit` but with raising the error)